### PR TITLE
Feature/chat mentors

### PR DESCRIPTION
**Pull request by androolloyd**
##### Description
Chat has moved into a flyout sidebar.

Sponsors can now set chat mentors for their bounties through the hackathon integrated view.

### CODE REVIEW

**Comment by danlipert** [comment]
@androolloyd looks like there are some conflicts, can you fix and rebase? thanks!

**Comment by danlipert** [comment]
oh also please put against master

File: chatdata/config/config.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "ServiceSettings": {
-        "SiteURL": "http://localhost:8065",
+        "SiteURL": "https://chat.androolloyd.com",
```
Comment (octavioamu) [inline chatdata/config/config.json:3]: mmm is this safe for production?

Comment (androolloyd) [reply chatdata/config/config.json:3]: doesn't get used in production, is only a development file anyways, will be removed.

File: app/retail/templates/shared/head.html [hunk 2]
```diff
@@ -76,33 +76,8 @@
...
-    };
-
-    socket.onclose = function(event) {
-      this.removeEventListener('close', socket.onclose);
-      spinner.addClass('d-none');
-      login_button.removeClass('d-none');
-    };
-  };
-</script>
+<script async src='{% static "v2/js/metamask-approval.js" %}'></script>
```
Comment (octavioamu) [inline app/retail/templates/shared/head.html:79]: why this is back?

Comment (androolloyd) [reply app/retail/templates/shared/head.html:79]: a regression from the rebase, i will remove

File: app/dashboard/views.py [hunk 3]
```diff
@@ -1144,6 +1145,55 @@ def previous_worked():
...
 
+@require_POST
+def bounty_mentor(request):
+
+    if not request.user.is_authenticated:
+        return JsonResponse({'message': 'Request Data invalid'}, status=400)
+
+    body_unicode = request.body.decode('utf-8')
+    body = json.loads(body_unicode)
+    can_manage = request.user.is_authenticated and any(
```
Comment (danlipert) [inline app/dashboard/views.py:1156]: dont need to check auth again

**Comment by owocki** [comment]
what is the ticket associated with this?  is this still an active PR?

File: app/dashboard/templates/dashboard/index-vue.html [hunk 4]
```diff
@@ -833,8 +873,14 @@ <h4 class="font-bigger-1 font-weight-bold mb-3">[[hackathon.name]] Wall of Fame<
       document.hackathonObj = JSON.parse(document.getElementById('hackathon-object').textContent);
       document.prizeFounders = JSON.parse(document.getElementById('prize-founders').textContent);
       document.activePanel = parseInt({{panel | safe}});
+<<<<<<< HEAD
```
Comment (danlipert) [inline app/dashboard/templates/dashboard/index-vue.html:876]: some rebase cruft is left in here

File: app/dashboard/views.py [hunk 5]
```diff
@@ -1165,6 +1166,53 @@ def previous_worked():
 
     return JsonResponse(params, status=200, safe=False)
 
+@require_POST
+def bounty_mentor(request):
+
+    if not request.user.is_authenticated:
+        return JsonResponse({'message': 'Request Data invalid'}, status=400)
+
+    body_unicode = request.body.decode('utf-8')
```
Comment (danlipert) [inline app/dashboard/views.py:1175]: how come we have to decode here... typically we don't have to

File: app/retail/views.py [hunk 6]
```diff
@@ -956,6 +956,28 @@ def get_specific_activities(what, trending_only, user, after_pk, request=None):
...
         activities = activities.filter(keyword_filter | base_filter)
+    elif 'hackathon:' in what:
+        terms = what.split(':')
+        pk = terms[1]
+
+        if len(terms) > 2:
+            print(terms)
+            if terms[2] == 'tribe':
+                key = terms[3]
+                print(key)
```
Comment (danlipert) [inline app/retail/views.py:967]: leftover prints

File: chatdata/config/config.json [hunk 7]
```diff
@@ -263,7 +263,7 @@
         "Secret": "xhLXNqXnOlkUiGCDFJLxQJjPb3CltLZNFZtzOsLgFfvwbYY60kYWRRBmqyjzfrnNyT6WWdIbfQEPpU854K2UiofmXtybFD1QkfIR9VVQyK3ttJpET7GormIBjK9CQRap",
         "Id": "VkHFfFlxnVCMYtYCWRxphbvPQUJjmER4cWtwqe7Q",
         "Scope": "read",
-        "AuthEndpoint": "http://localhost:8000/login/github/?next=",
+        "AuthEndpoint": "https://dapp.androolloyd.com/login/github/?next=",
```
Comment (danlipert) [inline chatdata/config/config.json:266]: Looks like some leftover local changes here

**Comment by danlipert** [comment]
@octavioamu this look good to you? other than the conflicts?

**Comment by PixelantDesign** [comment]
yayyyyy

**Pull request merged**